### PR TITLE
[Feature][PyDolphinScheduler] Support MLflow task in pyds

### LINKED ISSUE #11923: [Feature][PyDolphinScheduler] Support MLflow task in pyds
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

part of https://github.com/apache/dolphinscheduler/issues/11906

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jieguangzhou**
## Purpose of the pull request

close:  #11923

- [x] add pyds MLFLOW task
- [x] add YAML example
- [x] add doc

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

### CODE REVIEW

**Comment by zhongjiajie** [comment]
restart the failing test

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+    def __init__(
+        self,
+        name: str,
+        model_uri: str,
+        mlflow_tracking_uri: str = DEFAULT_MLFLOW_TRACKING_URI,
+        deploy_mode: str = MLflowDeployType.DOCKER,
+        port: int = 7000,
+        cpu_limit: float = 1.0,
+        memory_limit: str = "500M",
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]:
```suggestion
        mlflow_tracking_uri: Optional[str] = DEFAULT_MLFLOW_TRACKING_URI,
        deploy_mode: Optional[str] = MLflowDeployType.DOCKER,
        port: Optional[int] = 7000,
        cpu_limit: Optional[float] = 1.0,
        memory_limit: Optional[str] = "500M",
```

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    }
+
+    def __init__(
+        self,
+        name: str,
+        repository: str,
+        mlflow_tracking_uri: str = DEFAULT_MLFLOW_TRACKING_URI,
+        experiment_name: str = "",
+        parameters: str = "",
+        version: str = "master",
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]: we should also add keyword `optional` here

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    def __init__(
+        self,
+        name: str,
+        data_path: str,
+        automl_tool: str = "flaml",
+        mlflow_tracking_uri: str = DEFAULT_MLFLOW_TRACKING_URI,
+        experiment_name: str = "",
+        model_name: str = "",
+        parameters: str = "",
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]: `optional` here

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    :param port: deploy port, default is 7000
+    :param cpu_limit: cpu limit, default is 1.0
+    :param memory_limit: memory limit, default is 500M
+    """
+
+    mlflow_task_type = MLflowTaskType.MLFLOW_MODELS
+
+    _task_custom_attr = {
+        "mlflow_tracking_uri",
+        "mlflow_task_type",
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]: I find out we have same attribute with this two parameter, should we add a `Base` class for mlflow just we do for DVC  task type?

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py (hunk 1 above)
Comment (jieguangzhou) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]: done

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py (hunk 4 above)
Comment (jieguangzhou) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]: done

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py (hunk 3 above)
Comment (jieguangzhou) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]: done

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py (hunk 2 above)
Comment (jieguangzhou) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/mlflow.py]: done

**Comment by jieguangzhou** [comment]
@zhongjiajie PTAL, thanks

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks

**Pull request merged**